src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandlerBase.cs

```diff
@@ -41,7 +41,7 @@ public async Task<IEnumerable<CodeAction>> GetCodeActionsAsync(Solution solution
             var codeRefactorings = await _codeRefactoringService.GetRefactoringsAsync(document, textSpan, cancellationToken).ConfigureAwait(keepThreadContext);
 
             var codeActions = codeFixCollections.SelectMany(c => c.Fixes.Select(f => f.Action)).Concat(
-                                codeRefactorings.SelectMany(r => r.Actions));
+                                codeRefactorings.SelectMany(r => r.CodeActions.Select(ca => ca.action)));
```

Comment:
i wish we could track where things are lossy between us and LSP.

Reply:
Yep, an issue tracking this info being lost with LSP should be opened.